src/osdc/Objecter.cc

```diff
@@ -3147,7 +3158,7 @@ int Objecter::_calc_target(op_target_t *t, bool any_change)
 	osd = t->acting[best];
       }
       t->osd = osd;
-    } else {
+    } else if (!t->force_shard) {
```

Comment:
As at the beginning we do have `if (t->force_shard)`, isn't the extra check _always-true_?
(rechecking suggests the value might have changed along the way)